### PR TITLE
feat: add Vietnamese (vi) locale

### PR DESCRIPTION
**Pull request by nhannht**
## Summary

This PR adds a full Vietnamese (`vi`) locale to chrono-node, bringing it to parity with the English locale in terms of parser and refiner coverage.

### What's included

**11 parsers:**
- `VIStandardParser` — `ngày D tháng M năm YYYY` (with/without `ngày` prefix, optional year)
- `VIMonthYearParser` — `tháng M năm YYYY` / `tháng M/YYYY`
- `VIYearParser` — standalone `năm YYYY` and bare `YYYY TCN` (BC years, no `năm` prefix required)
- `VICasualDateParser` — `hôm nay`, `hôm qua`, `ngày mai`, `ngày kia`, `bây giờ`
- `VICasualTimeParser` — `sáng`, `trưa`, `chiều`, `tối`, `đêm`, `nửa đêm`, `bình minh`, `sáng sớm`
- `VITimeExpressionParser` — `lúc/vào 7 giờ 30 phút chiều`, `15:30`
- `VIWeekdayParser` — `thứ hai`–`thứ bảy`, `chủ nhật` with `tới/sau/qua` modifiers
- `VITimeUnitAgoFormatParser` — `3 ngày trước`, `1 tháng qua`
- `VITimeUnitLaterFormatParser` — `2 tuần sau`, `3 ngày nữa`
- `VITimeUnitWithinFormatParser` — `trong vòng 2 giờ`
- `VITimeUnitCasualRelativeFormatParser` — `tuần này/trước/sau`, `tháng này/trước/sau` (number optional — bare unit words supported)

**3 refiners:**
- `VIMergeDateTimeRefiner` — merges date + time results separated by `lúc/vào`
- `VIMergeDateRangeRefiner` — merges date ranges separated by `–/đến/tới/và`
- `VIMergeWeekdayComponentRefiner` — standard weekday merge

**16 test files, 95 tests** covering standard dates, casual dates/times, time expressions, weekdays, slash dates, date ranges, month/year, year, time units (ago/later/within/casual-relative), strict mode, forward date, negative cases, and isCertain/toBeDate assertions.

### Design notes

- `YEAR_PATTERN` supports BC years via `TCN` suffix (trước Công nguyên); `VIYearParser` also handles bare `YYYY TCN` without a `năm` prefix (e.g. `"179 TCN"`)
- `parseYear()` calls `findMostLikelyADYear()` for short years (e.g. `1945` stays `1945`, `75` → `1975`)
- Meridiem mapping: sáng=AM (12 giờ sáng=midnight), trưa=noon (~11 AM–1 PM), chiều=15h PM, tối=19h PM, đêm=22h PM, nửa đêm=0h AM, bình minh/sáng sớm=6h AM
- `VITimeUnitCasualRelativeFormatParser` uses an optional-number pattern so bare unit words (`tuần này`, `tháng trước`) match without a numeric prefix; defaults to quantity 1
- `strictMode` constructor parameter on all relative unit parsers (consistent with DE/FR)
- Exports added to `src/index.ts`

### Bug fixes included (found during self-review)

- **`VITimeExpressionParser`**: `trưa` meridiem was incorrectly grouped with `chiều/tối/đêm` — `11 giờ trưa` returned 23:00 instead of 11:00 AM. Fixed by separating `trưa` handling: hour < 10 → PM (+12), hour 10-11 → AM (keep as-is), hour 12 → PM (noon)
- **`VITimeExpressionParser`**: `12 giờ sáng` returned noon instead of midnight — added `if (hour === 12) hour = 0` guard matching EN convention
- **`VITimeExpressionParser`**: added minute validation — reject minute >= 60 (matching `AbstractTimeExpressionParser` behavior)
- **`VICasualTimeParser`**: removed redundant `\b` from PATTERN — JS `\b` fails on Vietnamese `đ` (non-ASCII), silently breaking standalone `đêm` parsing. `AbstractParserWithWordBoundaryChecking` already provides left-boundary via `(\W|^)`
- **`VIWeekdayParser`**: modifier group was non-capturing, making `qua` (last) undetectable; fixed to capturing group; also corrected `quả` typo → `qua`
- **`VITimeUnitCasualRelativeFormatParser`**: bare `tuần này`/`tháng trước` never matched due to required numeric prefix in `TIME_UNITS_PATTERN`; replaced with optional-number pattern
- **`package-lock.json`**: reverted to upstream — removed unrelated lockfile drift (dayjs removal, peer dependency changes)

🤖 Generated with [Claude Code](https://claude.ai/claude-code)

### CODE REVIEW

File: test/vi/vi_corpus.test.ts
Comment (wanasit) [inline test/vi/vi_corpus.test.ts:1]: I don't think this test and its fixtures suit the project. 

- It doesn't test specific patterns we want to support.
- Ensure that it's 85%+ accuracy is a corpus is not useful, and would be difficult to debug if failed.
- Embedding a corpus also make the project too large.

File: tsconfig.build.json [hunk 1]
```diff
@@ -5,6 +5,7 @@
   ],
   "compilerOptions": {
     "outDir": "dist/cjs",
-    "module": "commonjs"
+    "module": "commonjs",
+    "typeRoots": ["./node_modules/@types"]
```
Comment (wanasit) [inline tsconfig.build.json]: Please do not modify the compiler option and dependencies in the same commit. If you think this necessary or good for the project, please create a separate CL.

**Review by wanasit — COMMENTED** [review]
Thanks for the change. Could you remove some of the changes I mentioned in the comment?

File: test/vi/vi_corpus.test.ts
Comment (nhannht) [reply test/vi/vi_corpus.test.ts:1]: Thank you for the feedback — completely agree on all three points.

Removed `test/vi/vi_corpus.test.ts` and the entire `test/vi/fixtures/` directory (19 Wikipedia articles + curated JSON). The corpus added ~13k lines and wasn't testing specific parser patterns.

The VI locale now has 92 targeted unit tests across 16 suites, covering all 11 parsers and 3 refiners — including strict mode, `forwardDate`, `isCertain()` assertions, and negative cases. All tests follow the same `testSingleCase`/`testUnexpectedResult` conventions used by the EN locale.

File: tsconfig.build.json (hunk 1 above)
Comment (nhannht) [reply tsconfig.build.json]: Done — reverted `tsconfig.build.json` to match upstream. The `typeRoots` change was needed to resolve a build issue on my end, but I understand the preference to keep compiler/dependency changes in a separate CL. I'll submit that separately if needed.

**Comment by nhannht** [comment]
Hi @wanasit, thank you for the review. I've addressed both comments:

1. **Corpus test removed** — deleted `test/vi/vi_corpus.test.ts` and all `test/vi/fixtures/` (19 Wikipedia `.txt` files + curated JSON). Replaced with 92 targeted unit tests across 16 suites that test specific patterns, following the same `testSingleCase`/`testUnexpectedResult` conventions as the EN locale.

2. **`tsconfig.build.json` reverted** — restored to match upstream exactly. Will submit compiler changes as a separate CL if needed.

Additionally:
- Rebased on `upstream/master` to incorporate the Finnish locale merge (#642)
- Added strict mode, `forwardDate`, `isCertain()`, and expanded negative case tests to match EN test quality

All 92 tests pass. Will mark as ready for review once I've done a final check.

**Comment by nhannht** [comment]
Pushed two additional commits addressing issues found during self-review:

**`86f0076` fix(vi): fix trưa meridiem bug, \b word boundary bug, revert lockfile drift**
- `VITimeExpressionParser`: "trưa" was incorrectly grouped with "chiều/tối/đêm", causing `11 giờ trưa` to return 23:00 instead of 11:00 AM
- `VICasualTimeParser`: removed redundant `\b` from regex — JS `\b` doesn't work with Vietnamese `đ` (non-ASCII), silently breaking standalone `đêm` parsing
- Reverted `package-lock.json` to match upstream (unrelated lockfile drift)

**`8a476a6` test(vi): add minute validation, toBeDate assertions, strict/forward edge cases**
- Added minute >= 60 rejection in `VITimeExpressionParser` (matching `AbstractTimeExpressionParser`)
- Added `toBeDate()` assertions to casual tests (matching EN conventions)
- Added strict mode slash date acceptance test, forwardDate same-weekday edge case, impossible minute negative test

Test count: **95 tests across 16 suites**, all passing. Full suite (610 tests / 130 suites) also clean.

**Comment by wanasit** [comment]
Sorry for the delay. Thanks again for the improvement.
